src/sad_chan.F90

```diff
@@ -254,7 +254,7 @@ function create_sad_filename2(Ctrl, chan_num, SAD_Dir, LUTClass, crp_name) &
          stop error_stop_code
       end if
    else if (InstName(1:10) == 'SEVIRI-MSG') then
-      InstName = 'SEVIRI-MSG' // InstName(12:12)
+      InstName = 'SEVIRI-MSG' // InstName(11:12)
```

Comment:
Apologies, it turns out I had a slightly older version of orac utils to generate the driver files, so the platform and instrument name didn't have the hyphen. I suspect the reason you're getting the Read_LwRTM() and Read_SwRTM() is because I added an extra if statement (because of the same issue with the instrument name) at line 130-132 for read_lwrtm.F90, line 122-124 for read_swrtm.F90. I've now fixed my driver file generation code and the code runs after removing those edits.

Reply:
I'll have another go once you push those changes.